src/experimental/tcp_stream.jl

```diff
@@ -149,10 +149,6 @@ function Base.push!(stream::TCPStream, iheader::IPv4Header, packet::TCPPacket)
             isempty(packet.payload) && return
             @warn "TCP retransmission" src_endpoint dst_endpoint iheader packet
             stream.event_handler(src_endpoint.addr, dst_endpoint.addr, TCPE_RETRANSMIT)
-            if (theader.seq_num + length(packet.payload)) % UInt32 != src_endpoint.seq
```

Comment:
Should it not warn if the retransmit goes beyond the current sequence number?

Reply:
> It would be an unusual situation, but not strictly invalid -- TCP stack could have some data in send buffer and decide to combine it with retransmit data. I can leave a warning in place.

This could also happen if some data we've seen and some data was genuinely lost, so not seen by us before. And it's all been retransmitted in a single packet.

Reply:
I'm just concerned we run into a scenario where we don't handle an overlength retransmit (we should pad up the stream, do we?), and silently ignore it. I get ignoring the retransmits we've already seen.

Reply:
Does it look ok now?

Reply:
I guess, I haven't reviewed the logic of this, should be tested to know for sure, but I'll merge it as-is